basic-usage/recursive-mocking.md

```diff
@@ -1,267 +1,282 @@
-    [TestMethod]
-    [ExpectedException(typeof(StrictMockException))]
-    public void ShouldAssertNestedPropertySet()
-    {
-        // Arrange
-        var foo = Mock.Create<IFoo>(Behavior.Strict);
-
-        Mock.ArrangeSet(() => { foo.Bar.Value = 5; }).DoNothing();
-
-        // Act
-        foo.Bar.Value = 5;
-        foo.Bar.Value = 10; // This line will throw an exception.
-    }
-  {{endregion}}
-
-  #### __[VB]__
-
-  {{region RecursiveMocks#AssertNestedPropertySetSetups}}
-    <TestMethod()>
-    <ExpectedException(GetType(StrictMockException))>
-    Public Sub ShouldAssertNestedPropertySet()
-        ' Arrange
-        Dim foo = Mock.Create(Of IFoo)(Behavior.Strict)
-
-        Mock.ArrangeSet(Function() foo.Bar.Value = 5).DoNothing()
-
-        ' Act
-        foo.Bar.Value = 5
-        foo.Bar.Value = 10 ' This line will throws an exception.
-    End Sub
-  {{endregion}}
-
-We use `Bahavior.Strict` to enable only arranged calls and to reject any other calls. Only setting `foo.Bar.Value` to `5` is allowed and as we set it to `10`, an exception will be thrown. After assertion we actually set it to `5` and if we verify `foo` at that point an exception won't be thrown.
-
-## Nested Property and Method Calls
-We can call properties recursively, or we can do it with methods as well.
-
-  #### __[C#]__
-
-  {{region RecursiveMocks#AssertNestedPropertyHavingMethods}}
-    [TestMethod]
-    public void NestedPropertyAndMethodCalls()
-    {
-        // Arrange
-        var foo = Mock.Create<IFoo>();
-
-        Mock.Arrange(() => foo.Bar.Do("x")).Returns("xit");
-        Mock.Arrange(() => foo.Bar.Baz.Do("y")).Returns("yit");
-
-        // Act
-        var actualFooBarDo = foo.Bar.Do("x");
-        var actualFooBarBazDo = foo.Bar.Baz.Do("y");
-
-        // Assert
-        Assert.AreEqual("xit", actualFooBarDo);
-        Assert.AreEqual("yit", actualFooBarBazDo);
-    }
-  {{endregion}}
-
-  #### __[VB]__
-
-  {{region RecursiveMocks#AssertNestedPropertyHavingMethods}}
-    <TestMethod()>
-    Public Sub ShouldAssertNestedPropertyHavingMethods()
-        ' Arrange
-        Dim foo = Mock.Create(Of IFoo)()
-
-        Mock.Arrange(Function() foo.Bar.Do("x")).Returns("xit")
-        Mock.Arrange(Function() foo.Bar.Baz.Do("y")).Returns("yit")
-
-        ' Act
-        Dim actualFooBarDo = foo.Bar.Do("x")
-        Dim actualFooBarBazDo = foo.Bar.Baz.Do("y")
-
-        ' Assert
-        Assert.AreEqual("xit", actualFooBarDo)
-        Assert.AreEqual("yit", actualFooBarBazDo)
-    End Sub
-  {{endregion}}
-
-In this arrangement we specify that when calling `foo.Bar.Do` and `foo.Bar.Baz.Do` methods `"xit"` and `"yit"` should be returned respectively.
-
+---
+title: Mocking Chained Calls
+page_title: Recursive Mocking | JustMock Documentation
+description: Recursive Mocking
+previous_url: /basic-usage-recursive-mocking.html
+slug: justmock/basic-usage/recursive-mocking
+tags: recursive,mocking
+published: True
+position: 11
+---
+
+# Recursive Mocking
```

Comment:
The title in the menu on the left is changed but the title of the article, the description, and the wording inside are not. Could you check what is the preferred term while searching in google, pick that, and make SEO optimizations to the content?